### PR TITLE
Add new npm guide

### PR DESCRIPTION
**Pull request by olblak**
Add a new npm guide and refactor a bit the docker-compose and helm one

A preview environment is available to  see what the guide section looks like

### CODE REVIEW

File: assets/code_example/docs/guides/npm/updatecli.d/manifest.yaml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+      # The environment DRY_RUN is set and used by Updatecli to identify if:
+      #   1. we are in dry-run mode by running `updatecli diff --config manifest.yaml`
+      #   2. We are in apply mode by running `updatecli apply --config manifest.yaml`
+      command: npm install --package-lock-only --dry-run=$DRY_RUN axios@{{ source "npm" }}
+
+      # By default, shell command only have access to the DRY_RUN environment variable.
+      # But in this situation, we would also need the PATH value to identify where the npm command is
+      #
+      # As the PATH environment value is not defined, then it inherits the value from the Updatecli process.
```
Comment (lemeurherve) [inline assets/code_example/docs/guides/npm/updatecli.d/manifest.yaml:46]:
```suggestion
      # As the PATH environment value is not defined it inherits its value from the Updatecli process.
```

File: assets/code_example/docs/guides/npm/updatecli.d/manifest.yaml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      environments:
+        - name: PATH
+
+      # changedif defines how Updatecli should interpreted a failure or a change
```
Comment (lemeurherve) [inline assets/code_example/docs/guides/npm/updatecli.d/manifest.yaml:50]:
```suggestion
      # changedif defines how Updatecli should interpret a failure or a change.
```

File: assets/code_example/docs/guides/npm/updatecli.d/manifest.yaml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      # in this case we use the changedif of kind "file/checksum".
```
Comment (lemeurherve) [inline assets/code_example/docs/guides/npm/updatecli.d/manifest.yaml:51]:
```suggestion
      # In this case we use the "file/checksum" changedif kind.
```

File: assets/code_example/docs/guides/npm/updatecli.d/manifest.yaml [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+      # A filechecksum change would report a change.
```
Comment (lemeurherve) [inline assets/code_example/docs/guides/npm/updatecli.d/manifest.yaml:52]: I'm not sure to understand the subtilty here.

File: assets/code_example/docs/guides/npm/updatecli.d/manifest.yaml [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+      changedif:
+        kind: file/checksum
+        spec:
+          files:
+            - package-lock.json
+            - package.json
+
+      # npm expects to run the command from the directory containing the package.json
+      # workdir allows to specify the path where to run the command.
+      # If the resource has a scmid, then the workdir relative path is from the scm directory cloned
```
Comment (lemeurherve) [inline assets/code_example/docs/guides/npm/updatecli.d/manifest.yaml:62]:
```suggestion
      # If the resource has a scmid, then the workdir relative path is from the cloned scm directory
```

File: assets/code_example/docs/guides/npm/updatecli.d/manifest.yaml [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+      # in the temporary directory such as /tmp/updatecli on POSIX system.
+      workdir: "assets/code_example/docs/guides/npm/"
+
+# scms defines the git repository configuration which can then be used by the target and the action
```
Comment (lemeurherve) [inline assets/code_example/docs/guides/npm/updatecli.d/manifest.yaml:66]:
```suggestion
# scms defines the git repository configuration used by the target and the action.
```

File: assets/code_example/docs/guides/npm/updatecli.d/manifest.yaml [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    spec:
+      owner: updatecli
+      repository: website
+      # To handle secrets, Updatecli can either use environment variable or sops secrets.
+      # When executed from GitHub Action, using GITHUB_TOKEN and GITHUB_ACTOR is quite convenient.
+      token: '{{ requiredEnv "GITHUB_TOKEN" }}'
+      username: '{{ requiredEnv "GITHUB_ACTOR" }}'
+      branch: master
+
+# actions, are actions to execute if the state of a binded target is changed.
```
Comment (lemeurherve) [inline assets/code_example/docs/guides/npm/updatecli.d/manifest.yaml:80]:
```suggestion
# actions are actions to execute if the state of a binded target is changed.
```

File: assets/code_example/docs/guides/npm/updatecli.d/manifest.yaml [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+# such as opening a pullrequest.
+#
```
Comment (lemeurherve) [inline assets/code_example/docs/guides/npm/updatecli.d/manifest.yaml:82]:
```suggestion
# Ex: opening a pullrequest.
```

File: assets/code_example/docs/guides/npm/updatecli.d/manifest.yaml [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+actions:
+  default:
+    # The github/pullrequest means Updatecli will create a temporary branch containing all the changes
+    # generated by the different target using the same scmid.
```
Comment (lemeurherve) [inline assets/code_example/docs/guides/npm/updatecli.d/manifest.yaml:86]:
```suggestion
    # generated by the different targets using the same scmid.
```

File: assets/code_example/docs/guides/npm/updatecli.d/manifest.yaml [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+    #
+    # Then If needed, it opens a pullrequest targeting the scm branch referenced by scmid
```
Comment (lemeurherve) [inline assets/code_example/docs/guides/npm/updatecli.d/manifest.yaml:88]:
```suggestion
    # Then if there is a change, it will opens a pull request targeting the scm branch referenced by scmid.
```

File: content/en/docs/guides/docker-compose.adoc [hunk 11]
```diff
@@ -1,9 +1,9 @@
 ---
-title: "Guide: Docker Compose"
+title: "Guide: docker compose"
```
Comment (lemeurherve) [inline content/en/docs/guides/docker-compose.adoc:2]: All these modifications should be in another PR.

File: content/en/docs/guides/helm-chart.adoc [hunk 12]
```diff
@@ -1,9 +1,9 @@
 ---
-title: "Guide: Helm Chart"
+title: "Guide: helm chart"
```
Comment (lemeurherve) [inline content/en/docs/guides/helm-chart.adoc:2]: All these modifications should be in another PR.

File: content/en/docs/guides/npm.adoc [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+
+{{< ide_jsonschema >}}
+
+==== npm
+
+npm is a package manager for the Javascript programming language.
+
+==== GitHub PAT
+
+The GitHub PAT, is the token used to authenticate with the GitHub api
```
Comment (lemeurherve) [inline content/en/docs/guides/npm.adoc:48]:
```suggestion
The GitHub Personal Access Token is the token used to interact with the GitHub API
```

File: content/en/docs/guides/npm.adoc [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+link:https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token[Documentation]
+
+=== Pipeline
+==== Description
+
+Updatecli relies on one or multiple pipeline manifest to describe the expected Update pipeline.
```
Comment (lemeurherve) [inline content/en/docs/guides/npm.adoc:54]:
```suggestion
Updatecli relies on one or multiple pipeline manifests to describe the expected Update pipeline.
```
I don't understand the goal of this sentence, especially the "expected Update pipeline" (?)

File: content/en/docs/guides/npm.adoc [hunk 15]
```diff
@@ -0,0 +1,99 @@
...
+
+```
+export UPDATECLI_GITHUB_TOKEN=<your_PAT>
+export UPDATECLI_GITHUB_ACTOR=<your GH username>
+updatecli diff --config updatecli.d/manifest.yaml`
+```
+
+=== Going further
+
+Once happy with the Update pipeline, you can leverage CI environment to run it regularly. You'll need Updatecli installed in your CI environment, and the right credentials. More information link:https://www.updatecli.io/docs/automate/github_action/[here]
```
Comment (lemeurherve) [inline content/en/docs/guides/npm.adoc:84]:
```suggestion
Once you're happy with the Update pipeline, you can leverage CI environment to run it regularly.
You'll need Updatecli installed in your CI environment, and the right credentials.
More information link:https://www.updatecli.io/docs/automate/github_action/[in the corresponding GitHub Action documentation]
```

File: content/en/docs/guides/npm.adoc [hunk 16]
```diff
@@ -0,0 +1,99 @@
...
+
+This guide demonstrated a simple update pipeline, but real-world situations usually require either updating all dependencies specified in a package.json, or updating additional resources such as documentation, CI environment, etc.
+
+In the case of updating all dependencies from a package.json at once, the Autodiscovery feature is our current experimentation for addressing the problem. Feel free to reach out as we are still shaping this feature. To see what it looks like you can run `updatecli --experimental` from a repository containing package.json
```
Comment (lemeurherve) [inline content/en/docs/guides/npm.adoc:88]:
```suggestion
In the case of updating all dependencies from a package.json at once, the Autodiscovery feature is our current experimentation for addressing the problem.
Feel free to reach out as we are still shaping this feature.
To see what it looks like, you can run `updatecli --experimental` from a repository containing package.json
```

File: layouts/shortcodes/ide_jsonschema.html [hunk 17]
```diff
@@ -0,0 +1,13 @@
+<p>
```
Comment (lemeurherve) [inline layouts/shortcodes/ide_jsonschema.html:1]: All these modifications should be in another PR.

File: layouts/shortcodes/updatecli_description.html [hunk 18]
```diff
@@ -0,0 +1,17 @@
+<p>
```
Comment (lemeurherve) [inline layouts/shortcodes/updatecli_description.html:1]: All these modifications should be in another PR.

**Comment by olblak** [comment]
Thanks @lemeurherve for your feedback.
I received positive feedback for this so I am gonna merge it

File: content/en/docs/guides/docker-compose.adoc (hunk 11 above)
Comment (olblak) [reply content/en/docs/guides/docker-compose.adoc]: I agree with you but this time, it's gonna be here :D

File: assets/code_example/docs/guides/npm/updatecli.d/manifest.yaml [hunk 19]
```diff
@@ -0,0 +1,96 @@
...
+    # The default behavior for a target is to have one of its spec field default to the source resource output.
+    #
+    # More specifically for a resource of kind "shell", updatecli would append the value after the command such as
+    # `npm install --package-lock-only --dry-run=$DRY_RUN axios@ 1.0.0`
+    # Please note the space between axios@ and 1.0.0
+    #
+    # In this context, it doesn't work as the npm command requires axios@1.0.0 to be attached without a space
+    #
+    # Therefore we disable the default source output by using the parameter "disablesourceinput: true"
+    # And then we use the Updatecli specific templating function {{ source "<sourceID>" }}
```
[the remark this answers is not in the document]
Comment (olblak) [reply assets/code_example/docs/guides/npm/updatecli.d/manifest.yaml:27]: I agree, it's a specific edge case when used in the current situation. Not mentioning it may lead to  confusion. :/

File: assets/code_example/docs/guides/npm/updatecli.d/manifest.yaml (hunk 4 above)
Comment (olblak) [reply assets/code_example/docs/guides/npm/updatecli.d/manifest.yaml:52]: By default when using the shell resource, it return "✔" if the command output something on the console otherwise it return failure/warning

But it's possible to provide different kind of interpretation of the shell execution.

** file/checksum**

```
sources:
  success1:
    name: Test the source succeed according success criteria despite command failing
    kind: shell
    spec:
      command: "false"
      changedif:
        kind: file/checksum
        spec:
          files:
            - e2e/updatecli.d/success.d/shell/checksum.yaml
```
monitor the file checksum of `e2e/updatecli.d/success.d/shell/checksum.yaml` and if changed after the command execution then flag the resource execution has "changed"

There is also `exitcode` which allows to monitor command execution exit code
